File: t/sample/module1.c
#include <stdio.h>
#include "module2.h"
#include "module3.h"

int main() {
  int localvar = 0;
  void (*f)();

  say_hello();
  say_bye();

  printf("variable = %d\n", variable);
  variable = 20;

  printf("variable = %d\n", variable);
  printf("localvar = %d\n", localvar);

  f = callback;
  f();

  return 0;
}

File: t/samples/sample_basic/python/module1.py
from module2 import *
import module3

def main():
    localvar = 0

    say_hello()
    say_bye()

    print(f"variable = {module3.variable}")
    variable = 20

    print(f"variable = {variable}")
    print(f"localvar = {localvar}")

    module3.callback()
    return 0

main()